src/data-structures/disjoint-set/DisjointSet.js

```diff
@@ -1,97 +1,46 @@
-import DisjointSetItem from './DisjointSetItem';
-
 export default class DisjointSet {
   /**
-   * @param {function(value: *)} [keyCallback]
+   * @param {number} [size] - Number of disjoint elements in set
    */
-  constructor(keyCallback) {
```

Comment:
It is really good to have a possibility to put any objects in Disjoint Set. If we won't have this `keyCallback` it won't be possible. I understand that it makes Disjoint Set implementation more verbose but from the other side it makes the usage of Disjoint Set class less verbose and more convenient since I don't need to work with indices but rather with objects directly.

Reply:
After experimenting with creating a disjoint set that takes in any object, I now find your objections to be conflicted.  With this implementation, you object to forcing the user to create a mapping from arbitrary inputs to indices.  However, with the previous implementation, you are fully on board with forcing the user to create a mapping from arbitrary objects to strings.

I fail to see how these are different except that in the previous implementation the requirements around the mapping function are not documented.  For instance, a mapping function from arbitrary inputs to unique objects will fail!  This was not at all obvious to me.  I am now very convinced that, even if you don't think that the previous implementation requires a change, the key function should be much more meticulously documented.